resource_customizations/argoproj.io/ApplicationSet/health.lua

```diff
@@ -0,0 +1,22 @@
+hs = {}
+
+if obj.status ~= nil then
+  if obj.status.conditions ~= nil then
+    for i, condition in pairs(obj.status.conditions) do
+      if condition.type == "ErrorOccurred" and condition.status == "True" then
+        hs.status = "Degraded"
+        hs.message = condition.message
+        return hs
+      end
+      if condition.type == "ResourcesUpToDate" and condition.status == "True" then
+        hs.status = "Healthy"
+        hs.message = condition.message
+        return hs
+      end
+    end
+  end
+end
+
+hs.status = "Unknown"
+hs.message = ""
```

Comment:
By defaulting to Unknown, we'll break anyone using AppSets v0.2 and below (which don't contain status conditions). I think the default should be Healthy instead (with the caveat that we'll eventually need to drop support for v0.2 and below in the future).

Assuming we only support v0.3 and above, the default should be `Progressing` and not `Unknown` since an empty status just means that the ApplicationSet controller has not observed this object yet.

Reply:
Great point... another option would be to introduce a `.spec.statusVersion` field so we could toggle to the appropriate behavior for each version (or, even without that field, introduce some version sniffing logic).

For now I'll just default to Healthy. Let me know if you'd prefer some type of version specification/sniffing.